### PR TITLE
Easy-2791: relation exceptions

### PR DESCRIPTION
**Pull request by jo-pol**
Fixes Easy-2791: relation exceptions

#### When applied it will...
* 
* 
* 

#### How should this be manually tested?

run DdmSpec with logging switched on
https://github.com/DANS-KNAW/easy-fedora-to-bag/blob/81cdc0134c8d59d5b0d9f73563e076d1d2f15054/src/test/resources/logback.xml#L15

#### Related pull requests on github

repo                       | PR                | note
-------------------------- | ----------------- | ----
easy-                      | [PR#](PRlink)     | ..

### CODE REVIEW

File: src/test/scala/nl/knaw/dans/easy/fedoratobag/DdmSpec.scala [hunk 1]
```diff
@@ -792,6 +792,72 @@ class DdmSpec extends TestSupportFixture with EmdSupport with AudienceSupport wi
...
+      emdRights,
+    ))
+    val triedDDM = DDM(emd, Seq("D13200"), abrMapping)
+    triedDDM.map(normalized) shouldBe Success(normalized(
+      <ddm:DDM xsi:schemaLocation={ schemaLocation }>
+        { ddmProfile("D13200") }
+        <ddm:dcmiMetadata>
+          <ddm:references href="http://persistent-identifier.nl/urn:nbn:nl:ui:13-svxg-8g">Archeologisch onderzoek verbreding Hunzeloop Elzemaat</ddm:references>
+          <ddm:isPartOf>Second Timothy: When and Where? Text and Traditions in the Subscriptions</ddm:isPartOf>
+          <ddm:isPartOf>Briefrapport</ddm:isPartOf>
```
Comment (lindareijnhoudt) [inline src/test/scala/nl/knaw/dans/easy/fedoratobag/DdmSpec.scala:827]: i expected this should be `dc:isPartOf`, but apparently the `ddm:isPartOf` does not require the `scheme` attribute

File: src/test/scala/nl/knaw/dans/easy/fedoratobag/DdmSpec.scala [hunk 2]
```diff
@@ -792,6 +792,72 @@ class DdmSpec extends TestSupportFixture with EmdSupport with AudienceSupport wi
...
+          <eas:isPartOf>
+              <eas:subject-title>2005-09/11</eas:subject-title>
+              <eas:subject-link/>
+          </eas:isPartOf>
+          <eas:isPartOf>
+              <eas:subject-title>blabla</eas:subject-title>
+          </eas:isPartOf>
+          <eas:references>
+              <eas:subject-title>Archeologisch onderzoek verbreding Hunzeloop Elzemaat</eas:subject-title>
+              <eas:subject-link>urn:nbn:nl:ui:13-svxg-8g</eas:subject-link>
```
Comment (lindareijnhoudt) [inline src/test/scala/nl/knaw/dans/easy/fedoratobag/DdmSpec.scala:815]: let's also add one with a doi? like 
```
<eas:subject-link>10.17026/dans-something</eas:subject-link>
```

which then would result in:

```
<ddm:references href="https://doi.org/10.17026/dans-something">Archeologisch onderzoek verbreding Hunzeloop Elzemaat</ddm:references>
```

File: src/main/scala/nl/knaw/dans/easy/fedoratobag/DDM.scala [hunk 3]
```diff
@@ -311,10 +311,11 @@ object DDM extends DebugEnhancedLogging {
   }.getOrElse(notImplemented(s"relation ($key)")(rel))
 
   private def toHref(uri: URI) = Option(uri.getScheme).map {
-    case "urn" => "http://persistent-identifier.nl/" + uri.toString
-    case "http" | "https" => uri.toString
+    case "urn" => "https://persistent-identifier.nl/"
+    case "doi" => "https://doi.org/10.17026/"
```
Comment (lindareijnhoudt) [inline src/main/scala/nl/knaw/dans/easy/fedoratobag/DDM.scala:315]: doesn't this double the 10.17026? or is that tested for?

File: src/test/scala/nl/knaw/dans/easy/fedoratobag/DdmSpec.scala [hunk 4]
```diff
@@ -792,6 +792,77 @@ class DdmSpec extends TestSupportFixture with EmdSupport with AudienceSupport wi
...
+      </emd:relation>,
+      emdRights,
+    ))
+    val triedDDM = DDM(emd, Seq("D13200"), abrMapping)
+    triedDDM.map(normalized) shouldBe Success(normalized(
+      <ddm:DDM xsi:schemaLocation={ schemaLocation }>
+        { ddmProfile("D13200") }
+        <ddm:dcmiMetadata>
+          <ddm:references href="https://persistent-identifier.nl/urn:nbn:nl:ui:13-svxg-8g">Archeologisch onderzoek verbreding Hunzeloop Elzemaat</ddm:references>
+          <ddm:references>Acabadabra</ddm:references>
```
Comment (jo-pol) [inline src/test/scala/nl/knaw/dans/easy/fedoratobag/DdmSpec.scala:830]: link is missing...

**Pull request merged**